packages/argument-parser/src/lib/parsers/Args.ts

```diff
@@ -0,0 +1,750 @@
+import type { ChannelTypes, GuildBasedChannelTypes } from '@sapphire/discord.js-utilities';
+import { container } from '@sapphire/pieces';
+import { err, isErr, isOk, isSome, maybe, ok, type Err, type Maybe, type Ok, type Result } from '@sapphire/result';
+import type {
+	CategoryChannel,
+	DMChannel,
+	GuildMember,
+	Message,
+	NewsChannel,
+	Role,
+	StageChannel,
+	TextChannel,
+	ThreadChannel,
+	User,
+	VoiceChannel
+} from 'discord.js';
+import type * as Lexure from 'lexure';
+import { join } from 'path';
+import type { URL } from 'url';
+import { ArgumentError } from '../errors/ArgumentError';
+import { Identifiers } from '../errors/Identifiers';
+import { UserError } from '../errors/UserError';
+import type { EmojiObject } from '../resolvers/emoji';
+import type { Argument, IArgument } from '../structures/Argument';
+import { ArgumentStore } from '../structures/ArgumentStore';
+import type { Command } from '../structures/Command';
+
+container.stores.register(new ArgumentStore().registerPath(join(__dirname, '..', 'arguments')));
```

Comment:
Does stores actually exist outside framework? And even if so, nothing will load them 🤔 

I'd expect more of an ArgsStore.register('name', (parameter, whatever) => {}) (`register<T>(name: string, arg: IArgument<T>)`)

Reply:
I never used `pieces` so I don't know how this works. But I only checked the lib to know how can I use it (just to use it here 😄 . I'll go with the 2nd option.